oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospector.java

```diff
@@ -222,7 +227,17 @@ private OAuth2AuthenticatedPrincipal convertClaimsSet(Map<String, Object> claims
 			}
 			return v;
 		});
-		return new OAuth2IntrospectionAuthenticatedPrincipal(claims, authorities);
+		return this.authenticationConverter.convert(new OAuth2IntrospectionAuthenticatedPrincipal(claims, authorities));
+	}
+
+	private Converter<OAuth2TokenIntrospectionClaimAccessor, OAuth2AuthenticatedPrincipal> defaultAuthenticationConverter() {
+		return (accessor) -> (OAuth2IntrospectionAuthenticatedPrincipal) accessor;
+	}
+
+	public void setAuthenticationConverter(
```

Comment:
Will you please add JavaDoc here that describes what this method is and when to use it? It would also be nice if the doc explained what the default authentication converter does. It should also have a `@since 6.3` attribute.